photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMapLoader.scala

```diff
@@ -27,21 +28,52 @@ import org.apache.spark.SparkContext
 class PalDBIndexMapLoader extends IndexMapLoader {
   private var _storeDir: String = null
   private var _numPartitions: Int = 0
+  private var _namespace: String = null
+
+  override def prepare(sc: SparkContext, params: IndexMapParams, namespace: String = IndexMap.GLOBAL_NS): Unit = {
+    val palDBParams = params match {
+      case p: PalDBIndexMapParams => p
+      case other =>
+        throw new IllegalArgumentException(s"PalDBIndexMapLoader requires a params object of type " +
+          s"PalDBIndexMapParams. ${other.getClass.getName}")
+    }
+
+    if (!palDBParams.offHeapIndexMapDir.isEmpty && palDBParams.offHeapIndexMapNumPartitions != 0) {
+      _storeDir = palDBParams.offHeapIndexMapDir.get
+      _numPartitions = palDBParams.offHeapIndexMapNumPartitions
+      _namespace = namespace
 
-  override def prepare(sc: SparkContext, params: Params): Unit = {
-    if (!params.offHeapIndexMapDir.isEmpty && params.offHeapIndexMapNumPartitions != 0) {
-      _storeDir = params.offHeapIndexMapDir.get
-      _numPartitions = params.offHeapIndexMapNumPartitions
       (0 until _numPartitions).foreach(i =>
-        sc.addFile(new Path(_storeDir, PalDBIndexMap.partitionFilename(i)).toUri().toString())
+        sc.addFile(getPath(sc, new Path(_storeDir, PalDBIndexMap.partitionFilename(i, namespace))))
```

Comment:
This is really nice.

@timyitong Have we faced any issues previously when the input paths were not well taken care of?